teamcity/resource_build_config.go

```diff
@@ -630,6 +675,62 @@ func flattenBuildStepCmdLine(s *api.StepCommandLine) map[string]interface{} {
 	return m
 }
 
+func flattenBuildFeatures(bfs []api.BuildFeature) ([]map[string]interface{}, error) {
+	var bfsToSave []map[string]interface{}
+	for _, bf := range bfs {
+		bfToSave := make(map[string]interface{})
+		cbf := bf.(*api.CommonBuildFeature)
+		bfToSave["type"] = cbf.Type()
+
+		if cbf.Properties() != nil && cbf.Properties().Count > 0 {
+			bfToSave["properties"] = make([]map[string]string, 0)
+			propertyList := bfToSave["properties"].([]map[string]string)
+			for _, property := range cbf.Properties().Items {
+				propertyMap := make(map[string]string)
+				propertyMap["name"] = property.Name
+				propertyMap["value"] = property.Value
+				propertyList = append(propertyList, propertyMap)
+			}
+		}
+
+		bfsToSave = append(bfsToSave, bfToSave)
+	}
+	return bfsToSave, nil
+}
+
+func expandBuildFeatures(list interface{}) ([]api.BuildFeature, error) {
+	out := make([]api.BuildFeature, 0)
+	in := list.([]interface{})
+	for _, i := range in {
```

Comment:
I see that this follows the existing conventions of the file, but in this case, I think that ends up making the code less clear. `i` typically conveys an index value, so IMO it's not a good choice for the value variable in a loop. Something like `rawBuildFeature` or `rawBF` might be a better choice here, and similarly `s` might be renamed to `expandedBF`. I think that would make the logic of this function more apparent.

Similarly, `in` above could be called `rawBuildFeatures` (or `rawBFs`).